daemon/state.go

```diff
@@ -144,28 +148,32 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) {
 		// not in a goroutine) because regenerateRestoredEndpoints must guarantee
 		// upon returning that endpoints are exposed to other subsystems via
 		// endpointmanager.
-		ep.Mutex.RLock()
+
+		if lockerr := ep.RLockAlive(); lockerr != nil {
```

Comment:
This can be an unconditional lock as the endpoint has not been inserted into the endpointmanager yet and cannot be manipulated.

Reply:
done